BEST_PRACTICE.md

```diff
@@ -0,0 +1,247 @@
+ - [Example `package.json` for executables](#example-packagejson-for-exacutables).
+ - [Making your own library package](#making-your-own-libraries).
+ - [Example `package.json` for libraries](#example-packagejson-for-libraries).
+ - [How to install/uninstall executables](#install-and-uninstall-executables).
+
+For instructions on installation, click [here](https://github.com/clibs/clib#installation).
+
+## How to use installed libraries for your project
+
+Lets say you have your project, with a typical directory tree like this:
+
+```
+your-project/
+│
+├── deps/
+│   ├── trim.c/
+│   │   ├── trim.h
+│   │   ├── trim.c
+│   │   └── package.json
+│   │
+│   ├── commander/
+│   │   ├─ commander.h
+│   │   ├─ commander.c
+│   │   └─ package.json
+│   │
+│   └── logger/
+│       ├── logger.h
+│       ├── logger.c
+│       └── package.json
+│
+├── LICENSE
+│
+├── Makefile
+│
+├── README.md
+│
+├── package.json
+│
+├── src/
+│   ├── main.c
+│   ├── function.c
+│   ├── function.h
+│   └── etc...
+│
+└── test.sh
+```
+
+`test.sh` is your test script. `src` is the directory where you source
+code is in. And `deps` is the directory where you libraries will be
+downloaded when you run `clib install <username/library>`.
+
+Knowing all of that, lets have a look at an example Makefile.
+
+### Example Makefile
+
+```makefile
+# your c compiler
+CC = gcc
+
+# where to install
+PREFIX = /usr/local/bin
+
+# your output file
+TARGET = project-name
+
+# your project name
+TARGET = your-project
+
+CFLAGS = -Ideps -Wall
+
+# all the source files
+SRC = $(wildcard src/*.c)
+SRC += $(wildcard deps/*/*.c)
+
+OBJS = $(SRC:.c=.o)
+
+.PHONY:
+all: $(TARGET)
+
+.PHONY:
+$(TARGET): $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $(TARGET) $(OBJS)
+
+.PHONY:
+%.o: %.c
+	$(CC) $(DEP_FLAG) $(CFLAGS) $(LDFLAGS) -o $@ -c $<
+
+.PHONY:
+clean:
+	rm -f $(OBJS)
+
+.PHONY:
+install: $(TARGET)
+	cp -f $(TARGET) $(PREFIX)
+
+.PHONY:
+uninstall: $(PREFIX)/$(TARGET)
+	rm -f $(PREFIX)/$(TARGET)
+```
+
+This is a basic Makefile, and should work for most of your projects.
+
+You *could* have your Makefile install the libraries upon running it, but you
+would only need to do that to get the latest version of the library(s), in this
+case you proboly don't want that. You typicly want yo get the lates stable version
+for that library. By having a `package.json` file in your project repo, you can
+specify what packages you need, and what version of that package. Now have a look
+at a example `package.json` file for your project: (executable package)
+
+### Example package.json for exacutable project
+
+```
+{
+  "name": "executable-name",
+  "version": "1.0.0",
+  "repo": "your-github-name/project-name",
+  "dependencies": {
+    "stephenmathieson/trim.c": "0.0.2",
+    "clibs/commander": "1.3.2",
+    "clibs/logger": "0.0.1",
+  },
+  "install": "make install",
+  "uninstall": "make uninstall"
+}
+```
+
+Starting from the top, `"name"` is your package name. `"version"` is your package version.
```

Comment:
Please remove the stray newlines.

Reply:
:frowning_face: :frowning_face: :frowning_face: :frowning_face: Fine, but its easier to read and suggest changes like this, the final markdown file will look the same when viewing it.